src/main/drivers/pwm_output_dshot_hal.c

```diff
@@ -61,33 +61,29 @@ void pwmWriteDshotInt(uint8_t index, uint16_t value)
 
     uint8_t bufferSize = loadDmaBuffer(motor, packet);
 
-    if (motor->timerHardware->output & TIMER_OUTPUT_N_CHANNEL) {
-        if (HAL_TIMEx_PWMN_Start_DMA(&motor->TimHandle, motor->timerHardware->channel, motor->dmaBuffer, bufferSize) != HAL_OK) {
-            /* Starting PWM generation Error */
-            return;
-        }
-    } else {
-        if (HAL_TIM_PWM_Start_DMA(&motor->TimHandle, motor->timerHardware->channel, motor->dmaBuffer, bufferSize) != HAL_OK) {
-            /* Starting PWM generation Error */
-            return;
-        }
+    if (DMA_SetCurrDataCounter(&motor->TimHandle, motor->timerHardware->channel, motor->dmaBuffer, bufferSize) != HAL_OK) {
+        /* DMA set error */
+        return;
     }
 }
 
 void pwmCompleteDshotMotorUpdate(uint8_t motorCount)
 {
     UNUSED(motorCount);
+    for (int i = 0; i < dmaMotorTimerCount; i++) {
+        /* Reset timer counter */
+        LL_TIM_SetCounter(dmaMotorTimers[i].timer, 0);
```

Comment:
ISR may be called after setting counted to 0. This should not affect DShot output (output will be synchronized to period), but it creates low-probability race condition. 
Keeping the counter running may be better option (load should occur on next compare match, actual output will be delayed due to shadow register)

Or use ATOMIC_SECTION to improve determinism